webextension/background.js

```diff
@@ -333,30 +347,62 @@ const themeManager = {
 };
 
 const tabPageCounter = {
-  counter: {},
-
-  initTabCounter(tab) {
-    if (tab.id in this.counter) {
+  counters: {},
+
+  initTabCounter(tab, why = "user-activated-tab") {
+    // When the user becomes active,
+    // we ONLY need to initialize the activity counter
+    if (tab.id in this.counters && why === "user-became-active") {
+      this.counters[tab.id].activity = {
+        "cookieStoreId": tab.cookieStoreId,
+        "pageRequests": 0
+      };
+      return;
+    }
+    if (tab.id in this.counters) {
       return;
     }
-    this.counter[tab.id] = {
+    this.counters[tab.id] = {};
+    this.counters[tab.id].tab = {
+      "cookieStoreId": tab.cookieStoreId,
+      "pageRequests": 0
+    };
+    this.counters[tab.id].activity = {
       "cookieStoreId": tab.cookieStoreId,
       "pageRequests": 0
     };
   },
 
-  sendTabCountAndDelete(tabId) {
-    browser.runtime.sendMessage({
-      method: "sendTelemetryPayload",
-      event: "page-requests-completed-per-tab",
-      userContextId: this.counter[tabId].cookieStoreId,
-      pageRequestCount: this.counter[tabId].pageRequests
-    });
-    delete this.counter[tabId];
+  sendTabCountAndDelete(tabId, why = "user-closed-tab") {
+    if (why === "user-closed-tab") {
+      browser.runtime.sendMessage({
+        method: "sendTelemetryPayload",
+        event: "page-requests-completed-per-tab",
+        userContextId: this.counters[tabId].tab.cookieStoreId,
+        pageRequestCount: this.counters[tabId].tab.pageRequests
+      });
+      // When we send the ping because the user closed the tab,
+      // delete both the 'tab' and 'activity' counters
+      delete this.counters[tabId];
+    } else if (why === "user-went-idle") {
+      browser.runtime.sendMessage({
+        method: "sendTelemetryPayload",
+        event: "page-requests-completed-per-activity",
+        userContextId: this.counters[tabId].activity.cookieStoreId,
+        pageRequestCount: this.counters[tabId].activity.pageRequests
+      });
+      // When we send the ping because the user went idle,
+      // only reset the 'activity' counter
+      this.counters[tabId].activity = {
+        "cookieStoreId": this.counters[tabId].tab.cookieStoreId,
+        "pageRequests": 0
+      };
+    }
   },
 
   incrementTabCount(tab) {
-    this.counter[tab.id].pageRequests++;
+    this.counters[tab.id].tab.pageRequests++;
```

Comment:
I don't **think** I'm counting the delta between tab lifetime and activity states.

The 2 page counters here are:

* "per-tab" counter
* "per-activity" counter

### per-tab
1. User opens container tab
2. User visits 3 different pages
3. User closes container tab
   * We send a `page-requests-completed-per-tab` ping with the value 3

### per-activity
1. User opens a container tab
2. User visits 1 page in the container tab
3. User never closes tab
4. User goes "idle"
   * We send a `page-requests-completed-per-activity` ping with the value 1